pkg/server/chaosd/network.go

```diff
@@ -36,24 +36,31 @@ func (networkAttack) Attack(options core.AttackConfig, env Environment) (err err
 		ipsetName string
 	)
 
-	if attack.NeedApplyIPSet() {
-		ipsetName, err = env.Chaos.applyIPSet(attack, env.AttackUid)
-		if err != nil {
-			return errors.WithStack(err)
+	switch attack.Action {
+	case core.NetworkDNSAction:
```

Comment:
That seems a bit weird. Why can't we use some `if attack.NeedApplyDNS()`?

Reply:
But certainly `DNSAttack` will have `attack.NeedApplyIpset() == false`, so the original way would work properly?

Reply:
Yes, it can work, but I think use switch can make the logic more clear

Reply:
That's ok. Rest lgtm